raster-foundry/raster-foundry
# Implement Annotation API

## Overview
This PR adds CRUDL endpoints for annotations according to the swagger spec.

All annotations (both read-one and list actions) output GeoJSON.
The GeoJSON feature-collections output by the list endpoint contain pagination info.

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~

### Notes

The GeoJSON output functionality can be implemented by any case class that extends `GeoJSONSerializable[T]` and implements a `toGeoJSONFeature: T` method.
The route handling will also need to apply `fromPaginatedResponseToGeoJson` to the resulting `PaginatedResponse`.

## Testing Instructions

 * Run migrations
 * With the right auth credentials POST and GET to the `/api/projects/<projectid>/annotations/` endpoint and ensure the results are what you would expect
 * GET, PUT, and DELETE to the `/api/projects/<projectid>/annotations/<annotationid>` endpoint

Closes #2338
